temoa/_internal/table_writer.py

```
@@ -81,13 +77,122 @@ def __init__(self, config: TemoaConfig, epsilon: float = 1e-5) -> None:
+                    row_id = violation[1]
+                    parent_table = violation[2]
+                    logger.error(
+                        "  Table %s (row %d violates FK to parent '%s'",
+                        table_name,
+                        row_id,
+                        parent_table,
+                    )
+                # Depending on strictness requirements, we could raise an error here.
+                # For now, we log ensuring the user knows the output data might be inconsistent.
+        except sqlite3.Error as e:
+            logger.error('Error during foreign key validation: %s', e)
```

Ruff (0.14.8)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)